src/scripts/main.js

```diff
@@ -1,3 +1,47 @@
 'use strict';
 
 // write code here
+
+const table = document.querySelector('table');
+const rows = document.querySelector('tbody');
+
+table.addEventListener('click', (e) => {
+  if (e.target.nodeName !== 'TH' || e.target.closest('thead') === null) {
```

Comment:
This condition checks if the clicked element is a 'TH' within the 'thead'. Ensure that the table structure always includes a 'thead' for this to work correctly.